File: src/main/kotlin/magicbook/gtlitecore/api/unification/material/infos/MaterialIconSetWithRenderer.java
package magicbook.gtlitecore.api.unification.material.infos;

import gregtech.api.unification.material.info.MaterialIconSet;
import magicbook.gtlitecore.client.renderer.IItemRenderer;
import magicbook.gtlitecore.client.renderer.IItemRendererManager;
import net.minecraft.util.ResourceLocation;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class MaterialIconSetWithRenderer extends MaterialIconSet implements IItemRenderer, IItemRendererManager
{

    private final IItemRendererManager rendererManager;

    public MaterialIconSetWithRenderer(@NotNull String name,
                                       IItemRendererManager rendererManager)
    {
        super(name);
        this.rendererManager = rendererManager;
    }

    public MaterialIconSetWithRenderer(@NotNull String name,
                                       @NotNull MaterialIconSet parentIconSet,
                                       IItemRendererManager rendererManager)
    {
        super(name, parentIconSet);
        this.rendererManager = rendererManager;
    }

    public MaterialIconSetWithRenderer(@NotNull String name,
                                       @Nullable MaterialIconSet parentIconSet,
                                       boolean isRootIconSet,
                                       IItemRendererManager rendererManager)
    {
        super(name, parentIconSet, isRootIconSet);
        this.rendererManager = rendererManager;
    }

    @Override
    public IItemRendererManager getRendererManager()
    {
        return this.rendererManager;
    }

    @Override
    public void onRendererRegistry(ResourceLocation location)
    {
        this.rendererManager.onRendererRegistry(location);
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/api/unification/material/info/MaterialIconSetWithRenderer.kt
package gregtechlite.gtlitecore.api.unification.material.info

import gregtech.api.unification.material.info.MaterialIconSet
import gregtechlite.gtlitecore.client.renderer.CustomItemRenderer
import gregtechlite.gtlitecore.client.renderer.ItemRendererManager
import net.minecraft.util.ResourceLocation

class MaterialIconSetWithRenderer : MaterialIconSet, CustomItemRenderer, ItemRendererManager
{

    private val rendererManager: ItemRendererManager

    constructor(name: String,
                rendererManager: ItemRendererManager
    ) : super(name)
    {
        this.rendererManager = rendererManager
    }

    constructor(name: String,
                parentIconSet: MaterialIconSet,
                rendererManager: ItemRendererManager
    ) : super(name, parentIconSet)
    {
        this.rendererManager = rendererManager
    }

    constructor(name: String,
                parentIconSet: MaterialIconSet?,
                isRootIconSet: Boolean,
                rendererManager: ItemRendererManager
    ) : super(name, parentIconSet, isRootIconSet)
    {
        this.rendererManager = rendererManager
    }

    override fun getRendererManager(): ItemRendererManager = this.rendererManager

    override fun onRendererRegistry(location: ResourceLocation?)
    {
        this.rendererManager.onRendererRegistry(location)
    }

}

